src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1103,16 +1103,7 @@ TableNamesSet DatabaseCatalog::tryRemoveLoadingDependenciesUnlocked(const Qualif
         /// For DROP DATABASE we should ignore dependent tables from the same database.
         /// TODO unload tables in reverse topological order and remove this code
         if (check_dependencies)
-        {
-            TableNames from_other_databases;
-            for (const auto & table : dependent)
-                if (table.database != removing_table.database)
-                    from_other_databases.push_back(table);
-
-            if (!from_other_databases.empty())
-                throw Exception(ErrorCodes::HAVE_DEPENDENT_OBJECTS, "Cannot drop or rename {}, because some tables depend on it: {}",
-                                removing_table, fmt::join(from_other_databases, ", "));
-        }
+            checkTableCanBeRemovedOrRenamedImpl(dependent, removing_table, is_drop_database);
```

Comment:
I think we can remove the above checks i.e. `if (check_dependencies && !is_drop_database)` and `!dependent.empty()` as `checkTableCanBeRemovedOrRenamedImpl` will take care of it.

Reply:
Yep, I forgot to remove it, thanks